### PR TITLE
Create docker-publish.yml

### PR DESCRIPTION
**Pull request by tmooney**
Trying out the Github Container Registry!  On merging this PR it should build a "latest" for us.  I suspect it won't go back and rebuild for existing tags, so we might have to trigger a rebuild for those another way.

### CODE REVIEW

File: .github/workflows/docker-publish.yml [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+      - name: Build image
+        run: docker build . --file Dockerfile --tag $IMAGE_NAME
+
+      - name: Log into GitHub Container Registry
+      # TODO: Create a PAT with `read:packages` and `write:packages` scopes and save it as an Actions secret `CONTAINER_REGISTRY_PAT`
+        run: echo "${{ secrets.CONTAINER_REGISTRY_PAT }}" | docker login https://ghcr.io -u ${{ github.actor }} --password-stdin
+
+      - name: Push image to GitHub Container Registry
+        run: |
+          IMAGE_ID=ghcr.io/${{ github.repository_owner }}/$IMAGE_NAME
```
Comment (chrisamiller) [inline .github/workflows/docker-publish.yml:56]: I assume the image should then be accessible at `ghcr.io/genome/fp_filter-cwl`?

File: .github/workflows/docker-publish.yml [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    if: github.event_name == 'push'
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Build image
+        run: docker build . --file Dockerfile --tag $IMAGE_NAME
+
+      - name: Log into GitHub Container Registry
+      # TODO: Create a PAT with `read:packages` and `write:packages` scopes and save it as an Actions secret `CONTAINER_REGISTRY_PAT`
```
Comment (chrisamiller) [inline .github/workflows/docker-publish.yml]: Is this todo still something that needs to be complete before a build will work?

**Review by chrisamiller — APPROVED** [review]
I don't fully understand all the nuts and bolts here, but it seems reasonable

File: .github/workflows/docker-publish.yml (hunk 2 above)
Comment (tmooney) [reply .github/workflows/docker-publish.yml]: Ah no--I did it that already; that's just in the template.

File: .github/workflows/docker-publish.yml (hunk 1 above)
Comment (tmooney) [reply .github/workflows/docker-publish.yml:56]: That's the idea!

File: .github/workflows/docker-publish.yml (hunk 2 above)
Comment (tmooney) [inline .github/workflows/docker-publish.yml]:
```suggestion
      # Uses a PAT with `read:packages` and `write:packages` scopes
```

**Review by chrisamiller — APPROVED** [review]
+1

**Pull request merged**